bundles/binding/org.openhab.binding.ipx800/src/main/java/org/openhab/binding/ipx800/internal/Ipx800Binding.java

```diff
@@ -0,0 +1,299 @@
+/**
+ * Copyright (c) 2010-2014, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.ipx800.internal;
+
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.openhab.binding.ipx800.Ipx800BindingProvider;
+import org.openhab.binding.ipx800.internal.Ipx800Config.Ipx800DeviceConfig;
+import org.openhab.binding.ipx800.internal.Ipx800GenericBindingProvider.Ipx800BindingConfig;
+import org.openhab.binding.ipx800.internal.command.Ipx800Port;
+import org.openhab.binding.ipx800.internal.command.Ipx800PortType;
+import org.openhab.binding.ipx800.internal.exception.Ipx800UnknownDeviceException;
+import org.openhab.binding.ipx800.internal.itemslot.Ipx800Consumption;
+import org.openhab.binding.ipx800.internal.itemslot.Ipx800ConsumptionPeriod;
+import org.openhab.binding.ipx800.internal.itemslot.Ipx800AstableSwitch;
+import org.openhab.binding.ipx800.internal.itemslot.Ipx800Counter;
+import org.openhab.binding.ipx800.internal.itemslot.Ipx800Dimmer;
+import org.openhab.binding.ipx800.internal.itemslot.Ipx800DoubleClic;
+import org.openhab.binding.ipx800.internal.itemslot.Ipx800Item;
+import org.openhab.binding.ipx800.internal.itemslot.Ipx800Mirror;
+import org.openhab.binding.ipx800.internal.itemslot.Ipx800OutputItem;
+import org.openhab.binding.ipx800.internal.itemslot.Ipx800SimpleClic;
+
+import org.openhab.core.binding.AbstractBinding;
+import org.openhab.core.binding.BindingProvider;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The ipx800 binding. Handle the creation of items object linked to the devices.
+ * 
+ * @author Seebag
+ * @since 1.7.0
+ */
+public class Ipx800Binding extends AbstractBinding<Ipx800BindingProvider>
+		implements ManagedService {
+
+	private static final Logger logger = LoggerFactory
+			.getLogger(Ipx800Binding.class);
+	/**
+	 * Store the all ipx800 devices DeviceName, Ipx800DeviceConnector
+	 */
+	private Map<String, Ipx800DeviceConnector> devices = new HashMap<String, Ipx800DeviceConnector>();
+
+	public Ipx800Binding() {
+	}
+
+	@Override
+	public void bindingChanged(BindingProvider prov, String itemName) {
+		if (prov.getItemNames().contains(itemName)) {
+			try {
+				createItem(itemName, prov);
+			} catch (Ipx800UnknownDeviceException e) {
+				logger.error("Item {} will be ignored", itemName);
+			}
+		} else {
+			logger.debug("Removing item {}", itemName);
+			for (String deviceName : devices.keySet()) {
+				for (Ipx800Port slot : devices.get(deviceName).getAllPorts()) {
+					slot.destroyItem(itemName);
+				}
+			}
+		}
+	}
```

Comment:
when working with BindingProviders in internalReceiveCommand there should be no necessity to listen to bindingChanged Events anymore